helm/cloud-provider-cloud-director/charts/cloud-provider-for-cloud-director/values.yaml

```diff
@@ -11,8 +11,10 @@ ccmDeployment:
       effect: NoSchedule
     - key: "CriticalAddonsOnly"
       operator: "Exists"
+    - key: node-role.kubernetes.io/control-plane
+      effect: NoSchedule
     - key: node-role.kubernetes.io/master
       effect: NoSchedule
   matchExpressions:
-    - key: node-role.kubernetes.io/master
-      operator: "Exists"
+    - key: node-role.kubernetes.io/control-plane

```

Comment:
will this work in older versions?

Reply:
Yes. 

**Nodes running 1.22**
```yaml
Labels:
    node-role.kubernetes.io/control-plane: ""
    node-role.kubernetes.io/master: ""
Taints:
  - effect: NoSchedule
    key: node-role.kubernetes.io/master
```

**Nodes running 1.24**
```yaml
Labels:
    node-role.kubernetes.io/control-plane: ""
Taints:
  - effect: NoSchedule
    key: node-role.kubernetes.io/master
  - effect: NoSchedule
    key: node-role.kubernetes.io/control-plane
```

Reply:
The taint part is OK but `matchExpressions` doesn't seem OK to me. We use it in nodeSelectors and the pod will not be scheduled in older versions. Not tested but that is my understanding from the code.

https://github.com/giantswarm/cloud-provider-cloud-director-app/blob/0e8b113e7e73b9ccf541229910502f0566531a43/helm/cloud-provider-cloud-director/charts/cloud-provider-for-cloud-director/templates/deployment.yaml#L52

Reply:
Both new and old versions have the `node-role.kubernetes.io/control-plane` label so I don't get why it won't work on older ones?

Reply:
And by old I mean that we won't support dinosaur versions that are old enough to not have the controlplane label anyway 😄

Reply:
Sorry. I though one of them exists on nodes at a time. It seems both exist for [1.20-1.24] versions. Then it is OK.